user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementDetailsComponent.java

```diff
@@ -535,28 +532,58 @@ private void setPooledNgsMeasurementDetails(
       //Todo Add measurement specific pooled properties once defined for NGS
     }
 
-    private void setPooledNgsSampleDetails(NGSMeasurement ngsMeasurement) {
-      Grid<SampleId> sampleDetailsGrid = new Grid<>();
-      //Todo Wire pooled sample specific metadata once defined for NGS
-      sampleDetailsGrid.addColumn(sampleId -> retrieveSampleById(sampleId).orElseThrow().label())
+    private void setPooledNgsSampleDetails(
+        Collection<NGSSpecificMeasurementMetadata> ngsSpecificMeasurementMetadata) {
+      Grid<NGSSpecificMeasurementMetadata> sampleDetailsGrid = new Grid<>();
+      sampleDetailsGrid.addColumn(
+              metadata -> retrieveSampleById(metadata.measuredSample()).orElseThrow().label())
           .setHeader("Sample Label")
-          .setTooltipGenerator(sampleId -> retrieveSampleById(sampleId).orElseThrow().label())
+          .setTooltipGenerator(
+              metadata -> retrieveSampleById(metadata.measuredSample()).orElseThrow().label())
           .setAutoWidth(true);
       sampleDetailsGrid.addColumn(
-              sampleId -> retrieveSampleById(sampleId).orElseThrow().sampleCode().code())
+              metadata -> retrieveSampleById(metadata.measuredSample()).orElseThrow().sampleCode()
+                  .code())
           .setHeader("Sample Id")
           .setTooltipGenerator(
-              sampleId -> retrieveSampleById(sampleId).orElseThrow().sampleCode().code())
-          .setAutoWidth(true);
-      sampleDetailsGrid.addColumn(sampleId -> ngsMeasurement.indexI5().orElse(""))
-          .setHeader("Index I5")
-          .setTooltipGenerator(sampleId -> ngsMeasurement.indexI5().orElse(""))
+              metadata -> retrieveSampleById(metadata.measuredSample()).orElseThrow().sampleCode()
+                  .code())
           .setAutoWidth(true);
-      sampleDetailsGrid.addColumn(sampleId -> ngsMeasurement.indexI7().orElse(""))
+      sampleDetailsGrid.addColumn(metadata -> {
+            if (metadata.index().isEmpty()) {
+              return "";
+            } else {
+              return metadata.index().get().indexI7();
+            }
+          })
           .setHeader("Index I7")
-          .setTooltipGenerator(sampleId -> ngsMeasurement.indexI7().orElse(""))
+          .setTooltipGenerator(metadata -> {
+            if (metadata.index().isEmpty()) {
+              return "";
+            } else {
+              return metadata.index().get().indexI7();
+            }
+          }).setAutoWidth(true);
+      sampleDetailsGrid.addColumn(metadata -> {
+            if (metadata.index().isEmpty()) {
+              return "";
+            } else {
+              return metadata.index().get().indexI5();
+            }
+          })
+          .setHeader("Index I5")
+          .setTooltipGenerator(metadata -> {
+            if (metadata.index().isEmpty()) {
```

Comment:
`Optional.map`

Reply:
I just noticed that this might not be the case as empty index neither is an Optional nor null. I initially read the `isEmpty()` as the `isEmpty()` method of the `Optional` class. It turned out it is the `isEmpty()` method of the `String` class.